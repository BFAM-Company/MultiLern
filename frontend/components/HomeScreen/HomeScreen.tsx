import React from 'react';
import { Dimensions } from 'react-native';
import { View, Text, SafeAreaView, StyleSheet, ImageBackground, Image, TouchableOpacity, Platform } from 'react-native';
import Button from '../Button/Button';



function HomeScreen(props: any) {
  return (
    <View style={styles.mainContainer}>
        <ImageBackground
            source= {require('./../../assets/gradientBackground.png')}
            style={styles.ImageBackground}
            resizeMode='cover'
            blurRadius={40}
        >
            <View style={styles.AccountCreateContainer}>
                <View style={styles.LogoContainer}>
                    <Image source={require('./../../assets/multilern-logo.png')} style={styles.LogoImage}/>
                    <Text style={styles.titleText}>MultiLern</Text>
                </View>
                <View style={styles.mainTextContainer}>
                    <Text style={styles.mainText}>Dołącz do nas i podnieść swoją naukę na wyższy poziom</Text>
                </View>
                <View style={styles.buttonsContainer}>
                    <Button
                            colors={['white']}
                            buttonAction={() => {props.pageSwitcher('SignUp')}}
                            icons={[require('./../../assets/googleIcon.png'), 
                                    require('./../../assets/apple-icon.png'),
                                    require('./../../assets/facebook-icon.png'),
                                    require('./../../assets/lock-alt.png')
                                    ]}>
                        Zarejestruj się
                    </Button>
                    <Button
                        colors={['white']}
                        buttonAction={() => {props.pageSwitcher('SignUp')}}
                        icons={[require('./../../assets/logIn-icon.png')]}
                    >
                        Zaloguj się
                    </Button>
                    <Button
                        colors={['rgb(33,33,43)']}
                        fontColor='white'
                        buttonAction={() => {props.pageSwitcher('SignUp')}}
                        icons={[require('./../../assets/guest-icon.png')]}
                    >
                        Kontynuj jako gość
                    </Button>
                </View>
            </View>
        </ImageBackground>
    </View>
  );
}

const PCRatio = Platform.OS === 'web' ? 0.4 : 1
const LogoSize = Dimensions.get('window').width*0.10*PCRatio;

const styles = StyleSheet.create({
    mainContainer:{
        width:'100%',
        height:'100%',
        backgroundColor: 'white'
    },
    ImageBackground:{
        width:'100%',
        height:'100%',
        display:'flex',
        
        alignItems:'center',
        objectFit: 'cover',
        ...Platform.select({
            web: {
                justifyContent:'center',
            },
            default: {
                justifyContent:'flex-end',
            }
        }),
        
    },
    AccountCreateContainer:{
        height:'50%',
        display:'flex',
        justifyContent:'space-around',
        ...Platform.select({
            web: {
                alignItems:'center',
                width: '75%'
            },
            default: {
                marginBottom:20,
                width:'100%',
            }
        }),
    }, 
    LogoContainer:{
        width:'100%',
        display:'flex',
        flexDirection:'row',
        justifyContent:'center',
        alignItems:'center',
    },
    LogoImage:{
        width:LogoSize,
        height:LogoSize,        
        borderRadius:5,
    },
    titleText:{
        fontSize:30,
        color:'rgb(33,33,43)',
        fontWeight:'600',
        margin:10
    },
    mainTextContainer:{
        width:'100%',
        display:'flex',
        alignItems:'center',
    },
    mainText:{
        width:'90%',
        fontSize:19,
        fontWeight:'700',
        color:'gray',
        textAlign:'center',
    },
    buttonsContainer:{
        ...Platform.select({
            web: {
                width: '50%'
            },
            default: {
                width: '100%'
            }
        }),
        display:'flex',
        justifyContent:'space-around',
        alignItems:'center',
        margin:10,
    }
})


export default HomeScreen;