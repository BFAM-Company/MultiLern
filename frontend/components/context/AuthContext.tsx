import AsyncStorage from "@react-native-async-storage/async-storage";
import React, { createContext, useState, ReactNode, useContext } from "react";
import { UserDataContext } from "./UserContext";

interface AuthState {
  accessToken: string | null;
  refreshToken: string | null;
  authenticated: boolean | null;
}

export interface AuthContextType {
  authState: AuthState;
  getAccessToken: () => string | null;
  setAuthState: (state: AuthState) => void;
  logout: () => Promise<void>;
}

const initialAuthState: AuthState = {
  accessToken: null,
  refreshToken: null,
  authenticated: false,
};

const AuthContext = createContext<AuthContextType | null>(null);
const { Provider } = AuthContext;

const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [authState, setAuthState] = useState<AuthState>(initialAuthState);
  const userContext = useContext(UserDataContext)


  const logout = async () => {
    await AsyncStorage.clear()
    setAuthState({
      accessToken: null,
      refreshToken: null,
      authenticated: false,
    });
    userContext?.setUserData({
      nickname: '',
      email: 'null',
      avatar: null,
      isLogged: false
    })
  };

  const getAccessToken = () => {
    return authState.accessToken;
  };

  return (
    <Provider
      value={{
        authState,
        getAccessToken,
        setAuthState,
        logout,
      }}
    >
      {children}
    </Provider>
  );
};

export { AuthContext, AuthProvider };
