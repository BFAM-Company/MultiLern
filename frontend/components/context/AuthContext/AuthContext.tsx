import AsyncStorage from "@react-native-async-storage/async-storage";
import React, { createContext, useState, ReactNode, useContext, Dispatch, SetStateAction } from "react";
import { UserDataContext } from "../UserContext/UserContext";

interface AuthState {
  accessToken: string | null;
  refreshToken: string | null;
  authenticated: boolean | null;
  isLoggingByGuest?: boolean | null;
}

export interface AuthContextType {
  authState: AuthState;
  getAccessToken: () => string | null;
  setAuthState: Dispatch<SetStateAction<AuthState>>;
  logout: () => Promise<void>;
}

const initialAuthState: AuthState = {
  accessToken: null,
  refreshToken: null,
  authenticated: false,
  isLoggingByGuest: false
};

const AuthContext = createContext<AuthContextType | null>(null);
const { Provider } = AuthContext;

const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [authState, setAuthState] = useState<AuthState>(initialAuthState);
  const userContext = useContext(UserDataContext)


  const logout = async () => {
    userContext?.setUserData({
      id: -1,
      nickname: '',
      email: 'null',
      avatar: null,
    })
    await AsyncStorage.clear()
    setAuthState({
      accessToken: null,
      refreshToken: null,
      authenticated: false,
      isLoggingByGuest: false
    });
  };

  const getAccessToken = () => {
    return authState.accessToken;
  };

  return (
    <Provider
      value={{
        authState,
        getAccessToken,
        setAuthState,
        logout,
      }}
    >
      {children}
    </Provider>
  );
};

export { AuthContext, AuthProvider };
