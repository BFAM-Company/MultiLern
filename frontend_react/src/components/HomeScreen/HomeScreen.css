.main-container {
  width: 100%;
  height: 100vh; /* Cała wysokość ekranu */
  background-color: white;
  display: flex;
  justify-content: center;
  align-items: center;
}

.image-background {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: url("/assets/gradientBackground.png") no-repeat center center;
  background-size: cover;
  position: relative;
}

.account-create-container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 60%; /* Mniejsza szerokość na większym ekranie */
  max-width: 600px; /* Ustawiamy maksymalną szerokość kontenera */
  height: 60%; /* Wysokość kontenera dostosowana do ekranu */
}

.logo-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
}

.logo-image {
  width: 100px;
  height: 100px;
  border-radius: 5px;
}

.title-text {
  font-size: 36px; /* Większy rozmiar czcionki dla lepszej widoczności */
  color: rgb(33, 33, 43);
  font-weight: 600;
  margin: 10px;
}

.main-text-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 20px 0;
}

.main-text {
  font-size: 22px; /* Większy rozmiar tekstu */
  font-weight: 700;
  color: gray;
  text-align: center;
  max-width: 90%; /* Ograniczanie szerokości tekstu */
}

.buttons-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  gap: 15px; /* Odstępy między przyciskami */
}

@media (max-width: 768px) {
  /* Dostosowanie do mniejszych ekranów */
  .account-create-container {
    width: 80%;
  }

  .title-text {
    font-size: 28px; /* Zmniejszenie rozmiaru czcionki na mniejszych ekranach */
  }

  .main-text {
    font-size: 18px; /* Zmniejszenie tekstu na mniejszych ekranach */
  }

  .buttons-container {
    width: 100%;
    flex-direction: column;
  }
}
