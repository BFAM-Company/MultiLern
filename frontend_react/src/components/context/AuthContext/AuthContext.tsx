import React, { createContext, useState, ReactNode, Dispatch, SetStateAction } from 'react';

interface AuthState {
  accessToken: string | null;
  refreshToken: string | null;
  authenticated: boolean | null;
  isLoggingByGuest?: boolean | null;
}

export interface AuthContextType {
  authState: AuthState;
  getAccessToken: () => string | null;
  setAuthState: Dispatch<SetStateAction<AuthState>>;
  logout: () => void;
}

const initialAuthState: AuthState = {
  accessToken: null,
  refreshToken: null,
  authenticated: false,
};

const AuthContext = createContext<AuthContextType | null>(null);

const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [authState, setAuthState] = useState<AuthState>(initialAuthState);

  const logout = () => {
    localStorage.clear();
    setAuthState(initialAuthState);
  };

  const getAccessToken = () => authState.accessToken;

  return (
    <AuthContext.Provider value={{ authState, getAccessToken, setAuthState, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export { AuthContext, AuthProvider };
