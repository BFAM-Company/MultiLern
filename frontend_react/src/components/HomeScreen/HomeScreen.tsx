import React, { useContext } from "react";
import { AuthContext } from "../context/AuthContext/AuthContext";
import "./HomeScreen.css";
import Button from "../Button/Button";
import { useNavigate } from "react-router-dom";


const HomeScreen = () => {
  const authContext = useContext(AuthContext);
  const navigate = useNavigate()

  const loginByGuest = async () => {
    authContext?.setAuthState({
      accessToken: "dupa",
      refreshToken: "dupa",
      authenticated: true,
      isLoggingByGuest: true,
    });
    navigate('/main')
  };

  return (
    <div className="main-container">
      <div className="image-background">
        <div className="account-create-container">
          <div className="logo-container">
            <img
              alt="Logo"
              src="/assets/multilern-logo.png"
              className="logo-image"
            />
            <h1 className="title-text">MultiLern</h1>
          </div>
          <div className="main-text-container">
            <p className="main-text">
              Dołącz do nas i podnieść swoją naukę na wyższy poziom
            </p>
          </div>
          <div className="buttons-container">
            <Button
              colors={["#ffffff"]}
              fontColor="rgb(255, 255, 255)"
              buttonAction={() => navigate("/signup")}
              icons={[
                "/assets/discord-icon.png",
                "/assets/apple-icon.png",
                "/assets/facebook-icon.png",
                "/assets/lock-alt.png",
              ]}
            >
              Zarejestruj się
            </Button>
            <Button
              colors={["#ffffff"]}
              fontColor="rgb(255, 255, 255)"
              buttonAction={() => navigate("/login")}
              icons={["/assets/logIn-icon.png"]}
            >
              Zaloguj się
            </Button>
            <Button
              colors={["rgb(33,33,43)"]}
              fontColor="rgb(255, 255, 255)"
              buttonAction={loginByGuest}
              icons={["/assets/guest-icon.png"]}
            >
              Kontynuuj jako gość
            </Button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default HomeScreen;
