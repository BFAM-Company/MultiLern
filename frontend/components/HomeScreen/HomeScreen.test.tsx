import React from 'react';
import { render, fireEvent, cleanup} from '@testing-library/react-native';
import HomeScreen from './HomeScreen';
import { AuthContext } from '../context/AuthContext/AuthContext';


jest.mock("@react-native-async-storage/async-storage", () => require("../../__mocks__/mock-async-storage"));

afterEach(cleanup);


describe('<HomeScreen />', () => {
  it('renders correctly', () => {
    const mockPageSwitcher = jest.fn();
    const { toJSON } = render(<HomeScreen pageSwitcher={mockPageSwitcher} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls pageSwitcher with SignUp when Zarejestruj się button is pressed', () => {
    const mockPageSwitcher = jest.fn();
    const { getByText } = render(<HomeScreen pageSwitcher={mockPageSwitcher} />);
    
    const registerButton = getByText('Zarejestruj się');
    fireEvent.press(registerButton);

    expect(mockPageSwitcher).toHaveBeenCalledWith('SignUp');
  });

  it('calls pageSwitcher with LogIn when Zaloguj się button is pressed', () => {
    const mockPageSwitcher = jest.fn();
    const { getByText } = render(<HomeScreen pageSwitcher={mockPageSwitcher} />);
    
    const loginButton = getByText('Zaloguj się');
    fireEvent.press(loginButton);

    expect(mockPageSwitcher).toHaveBeenCalledWith('LogIn');
  });

  it('calls loginByGuest function when Kontynuuj jako gość button is pressed', () => {
    const mockSetAuthState = jest.fn();
    const mockGetAccessToken = jest.fn(() => 'testAccessToken');
    const mockLogout = jest.fn();

    const mockPageSwitcher = jest.fn();


    const mockAuthContext = {
      setAuthState: mockSetAuthState,
      getAccessToken: mockGetAccessToken,
      logout: mockLogout,
      authState: {
        accessToken: '',
        refreshToken: '',
        authenticated: false,
        isLoggingByGuest: false,
      }
    };

    const { getByText } = render(
      <AuthContext.Provider value={mockAuthContext}>
        <HomeScreen pageSwitcher={mockPageSwitcher} />
      </AuthContext.Provider>
    );
    
    const guestButton = getByText('Kontynuuj jako gość');
    fireEvent.press(guestButton);

    expect(mockSetAuthState).toHaveBeenCalledWith({
      accessToken: 'dupa',
      refreshToken: 'dupa',
      authenticated: true,
      isLoggingByGuest: true
    });
  });

  it('renders logo and title correctly', () => {
    const mockPageSwitcher = jest.fn();

    const { getByText, getByTestId } = render(<HomeScreen pageSwitcher={mockPageSwitcher} />);
    
    expect(getByText('MultiLern')).toBeTruthy();
    expect(getByTestId('logoImage')).toBeTruthy();
  });

  it('renders main text correctly', () => {
    const mockPageSwitcher = jest.fn();

    const { getByText } = render(<HomeScreen pageSwitcher={mockPageSwitcher} />);
    
    expect(getByText('Dołącz do nas i podnieść swoją naukę na wyższy poziom')).toBeTruthy();
  });

});
