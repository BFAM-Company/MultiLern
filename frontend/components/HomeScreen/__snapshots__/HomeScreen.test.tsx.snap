// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "flex-end",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      blurRadius={40}
      resizeMode="cover"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "display": "flex",
          "height": "50%",
          "justifyContent": "space-around",
          "marginBottom": 20,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "borderRadius": 5,
              "height": 75,
              "width": 75,
            }
          }
          testID="logoImage"
        />
        <Text
          style={
            {
              "color": "rgb(33,33,43)",
              "fontSize": 30,
              "fontWeight": "600",
              "margin": 10,
            }
          }
        >
          MultiLern
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 19,
              "fontWeight": "700",
              "textAlign": "center",
              "width": "90%",
            }
          }
        >
          Dołącz do nas i podnieść swoją naukę na wyższy poziom
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-around",
            "margin": 10,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 20,
              "display": "flex",
              "elevation": 16,
              "flexDirection": "row",
              "height": 80.03999999999999,
              "justifyContent": "space-between",
              "margin": 15,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.44,
              "shadowRadius": 10.32,
              "width": "80%",
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            colors={
              [
                4294967295,
                4294967295,
              ]
            }
            style={
              {
                "alignItems": "center",
                "borderRadius": 20,
                "display": "flex",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 40,
                "width": "100%",
              }
            }
            testID="gradient"
          >
            <Text
              style={
                [
                  {
                    "color": "rgb(33,33,43)",
                    "fontSize": 17,
                    "fontWeight": "700",
                  },
                  {
                    "color": undefined,
                    "textAlign": "left",
                    "width": "auto",
                  },
                ]
              }
            >
              Zarejestruj się
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "margin": 10,
                }
              }
            >
              <Image
                source={1}
                style={
                  [
                    {
                      "height": 45,
                      "margin": 5,
                      "width": 45,
                    },
                    {
                      "tintColor": undefined,
                    },
                  ]
                }
                testID="account-service-icon"
              />
              <Image
                source={1}
                style={
                  [
                    {
                      "height": 45,
                      "margin": 5,
                      "width": 45,
                    },
                    {
                      "tintColor": undefined,
                    },
                  ]
                }
                testID="account-service-icon"
              />
              <Image
                source={1}
                style={
                  [
                    {
                      "height": 45,
                      "margin": 5,
                      "width": 45,
                    },
                    {
                      "tintColor": undefined,
                    },
                  ]
                }
                testID="account-service-icon"
              />
              <Image
                source={1}
                style={
                  [
                    {
                      "height": 45,
                      "margin": 5,
                      "width": 45,
                    },
                    {
                      "tintColor": undefined,
                    },
                  ]
                }
                testID="account-service-icon"
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 20,
              "display": "flex",
              "elevation": 16,
              "flexDirection": "row",
              "height": 80.03999999999999,
              "justifyContent": "space-between",
              "margin": 15,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.44,
              "shadowRadius": 10.32,
              "width": "80%",
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            colors={
              [
                4294967295,
                4294967295,
              ]
            }
            style={
              {
                "alignItems": "center",
                "borderRadius": 20,
                "display": "flex",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 40,
                "width": "100%",
              }
            }
            testID="gradient"
          >
            <Text
              style={
                [
                  {
                    "color": "rgb(33,33,43)",
                    "fontSize": 17,
                    "fontWeight": "700",
                  },
                  {
                    "color": undefined,
                    "textAlign": "left",
                    "width": "auto",
                  },
                ]
              }
            >
              Zaloguj się
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "margin": 10,
                }
              }
            >
              <Image
                source={1}
                style={
                  [
                    {
                      "height": 45,
                      "margin": 5,
                      "width": 45,
                    },
                    {
                      "tintColor": undefined,
                    },
                  ]
                }
                testID="account-service-icon"
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 20,
              "display": "flex",
              "elevation": 16,
              "flexDirection": "row",
              "height": 80.03999999999999,
              "justifyContent": "space-between",
              "margin": 15,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.44,
              "shadowRadius": 10.32,
              "width": "80%",
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            colors={
              [
                4280361259,
                4280361259,
              ]
            }
            style={
              {
                "alignItems": "center",
                "borderRadius": 20,
                "display": "flex",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 40,
                "width": "100%",
              }
            }
            testID="gradient"
          >
            <Text
              style={
                [
                  {
                    "color": "rgb(33,33,43)",
                    "fontSize": 17,
                    "fontWeight": "700",
                  },
                  {
                    "color": "white",
                    "textAlign": "left",
                    "width": "auto",
                  },
                ]
              }
            >
              Kontynuuj jako gość
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "margin": 10,
                }
              }
            >
              <Image
                source={1}
                style={
                  [
                    {
                      "height": 45,
                      "margin": 5,
                      "width": 45,
                    },
                    {
                      "tintColor": "white",
                    },
                  ]
                }
                testID="account-service-icon"
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </View>
</View>
`;
